README.md

```diff
@@ -117,13 +128,24 @@ Filters can be negated to exclude results:
 /v1/{type}?filter=metadata.name!=foo
 ```
 
-Arrays are searched for matching items. If any item in the array matches, the
+**If SQLite caching is disabled** (`server.Options.SQLCache=false`),
+arrays are searched for matching items. If any item in the array matches, the
 item is included in the list.
 
 ```
 /v1/{type}?filter=spec.containers.image=alpine
 ```
 
+**If SQLite caching is enabled** (`server.Options.SQLCache=true`),
+filtering is only supported for a subset of attributes:
+- `metadata.name`, `metadata.namespace` and `metadata.timestamp` for any resource kind
+- a short list of hardcoded attributes for a selection of specific types listed
+in [typeSpecificIndexFields](https://github.com/rancher/steve/blob/main/pkg/stores/sqlproxy/proxy_store.go#L52-L58)
+- the special string `metadata.fields[N]`, with N starting at 0, for all columns
+displayed by `kubectl get $TYPE`. For example `secrets` have `"metadata.fields[0]"`,
```

Comment:
i think these are crd additionalPrinterColumns (https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#additional-printer-columns), though not sure how those apply to built in resources. maybe too much detail for this doc?

Reply:
I am good with a bit more detail here, this is by devs for devs

Fixed in 2119538